### PR TITLE
fix cursor pointer to checkbox

### PR DESCRIPTION
**Pull request by garrettbear**
## **Description**
 Fix the cursor to be a pointer on the hover for `checkbox` component. See screen recordings 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23786?quickstart=1)

## **Related issues**

Fixes: 

## **Manual testing steps**

1. Run stoybook
2. View and hover [Checkbox component](http://localhost:6006/?path=/docs/components-componentlibrary-checkbox--docs)
3. Confirm you see the pointer cursor 

### **Before**

https://github.com/MetaMask/metamask-extension/assets/26469696/3bdca80e-ca5b-4090-b0e0-c74c76c775d7

### **After**

https://github.com/MetaMask/metamask-extension/assets/26469696/830e6c7b-c622-46b1-a841-413597d4e216

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/component-library/checkbox/checkbox.scss [hunk 1]
```diff
@@ -13,6 +11,7 @@
 
     &:hover:not(:disabled) {
       background-color: var(--color-background-default-hover);
+      cursor: pointer;
```
Comment (georgewrmarshall) [inline ui/components/component-library/checkbox/checkbox.scss:14]: Will this work when the label is present as well? We may need it in both places. Just waiting for storybook build to test

File: ui/components/component-library/checkbox/checkbox.scss [hunk 2]
```diff
@@ -1,6 +1,4 @@
 .mm-checkbox {
-  cursor: pointer;
```
Comment (georgewrmarshall) [inline ui/components/component-library/checkbox/checkbox.scss:2]: Will the label still use the pointer when it is present or just the checkbox? I think we may need it in both but haven't tested yet. Waiting for storybook build on PR

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Checkbox is good! But I think we will need to the cursor:pointer in both places

**Review by georgewrmarshall — APPROVED** [review]
LGTM 🚀 🚀 🚀

File: ui/components/component-library/checkbox/checkbox.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/checkbox/checkbox.scss]: fixed in [276b81c](https://github.com/MetaMask/metamask-extension/pull/23786/commits/276b81cb74dab753c466ee25a3300c8f80312e45)

**Review by darkwing — APPROVED** [review]
I love this attention to detail!

**Pull request merged**